course/src/Vector3.cc

```diff
@@ -0,0 +1,156 @@
+#include "Vector3.h"
+
+namespace ekumen
+{
+
+Vector3::Vector3(const double x, const double y, const double z)
+{
+    values[0] = x;
+    values[1] = y;
+    values[2] = z;
+}
+Vector3::Vector3()
+{
+    values[0] = 0.;
+    values[1] = 0.;
+    values[2] = 0.;
+}
+Vector3::Vector3(const std::initializer_list<double> &list){
+    values[0] = *(list.begin() + 0);
+    values[1] = *(list.begin() + 1);
+    values[2] = *(list.begin() + 2);
+}
+double Vector3::norm() const
+{
+    return sqrt(dot(Vector3(x(), y(), z())));
+}
+
+double Vector3::dot(const Vector3 &q) const
+{
+    return ((values[0] * q.x()) + (values[1] * q.y()) + (values[2] * q.z()));
```

Comment:
For readability's sake you can use  `(x() * q.x()) + (y() * q.y()) + (z() * q.z()) `.